Modules/feed/feed_controller.php

```diff
@@ -55,7 +55,7 @@ function feed_controller()
                 if ($feed->exist($feedid)) {  // if the feed exists
                    $f = $feed->get($feedid);
                    if ($f['public'] || ($session['userid']>0 && $f['userid']==$session['userid'] && $session['read'])) {
-                       $result[$i] = 1*$feed->get_value($feedid); // null is a valid response
+                       $result[$i] = $feed->get_value($feedid); // null is a valid response
```

Comment:
Just casting will not work because NULL values are valid data.
A null means engine had no data, so in graphics you know there is no data (a gap appears) instead of it joining the last data points.
Also virtual feeds make use of null and there are specific processors dealing with null.